### ISSUE TITLE
Foreign key conflict

### ISSUE THREAD

**Issue by chukwu** [jtheisen/reconciler#4]
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_EventTypeContents_EventTypes_EventTypeId". The conflict occurred in database "knottfar", table.

```
    public class EventType
    {
        [Key]
        public int EventTypeId { get; set; }

        public string Images { get; set; }
        public DateTime EntryDate { get; set; }

        public Customer EnteredBy { get; set; }

        public List<EventTypeContent> Translations { get; set; }

        public EventType()
        {
            EntryDate = MyDateTime.Now();
            Translations = new List<EventTypeContent>();
        }
    }

    public class EventTypeContent
    {
        [Key]
        public int EventTypeContentId { get; set; }

        public string Title { get; set; }
        public string Description { get; set; }
        public string Language { get; set; }
        public DateTime EntryDate { get; set; }

        public Customer EnteredBy { get; set; }

        public int EventTypeId { get; set; }
        public EventType EventType { get; set; }
    }
```

`EventType reconciled = await _appDbContext.ReconcileAsync(eventType, e => e.WithMany(m => m.Translations));`

Everything is fine when I make changes to the one/many translations or delete a collection. But when I add a new translation to the collection. hell breaks loose.

**Comment by chukwu** [comment]
If I add below before reconcile, it works but breaks deleted entity by not deleting it. No fatal error.

          _appDbContext.Attach(eventType);

            IEnumerable<EntityEntry> unchangedEntities = _appDbContext.ChangeTracker.Entries().Where(x => x.State == EntityState.Unchanged);

            foreach (EntityEntry ee in unchangedEntities)
            {
                ee.State = EntityState.Modified;
            }
            await _appDbContext.SaveChangesAsync();
```

**Comment by chukwu** [comment]
**I found the issue.** 

It's my fault BUT......I feel Reconciler should have handled it better.

My angular app was sending adding zero(0) to the foreign key field. For some reason though, entity framework handles it well and adds the new record anyways without any FOREIGN KEY conflict.

**Comment by jtheisen** [comment]
I don't yet understand the whole picture.

Can you give me the entity template you reconciled against that was giving the error?

In particular, what does "sending adding zero(0) to the foreign key field" mean?

And are you using EF6 or EF Core?

**Comment by chukwu** [comment]
First, I'm using EFCore.
Looking at the model or DBSet class above, you would see that I have two related tables. One to many relationship. 

1. Add new **EventTypeContent** to an existing record of **EventType**  and set EventTypeId field to zero.

I wish i can be more elaborate.

**Comment by jtheisen** [comment]
So the behavior you see is just what you'd get if you do the same thing without Reconciler, correct?

**Comment by chukwu** [comment]
No!!! I introduced Reconcile because I needed to track deleted data. Before introducing Reconciler, I could only add and edit parent(**EventType**) its child/collections(**EventTypeContent**). No Exception!!!

My code was just: 


```
        IEnumerable<EntityEntry> unchangedEntities = _appDbContext.ChangeTracker.Entries().Where(x => x.State == EntityState.Unchanged);

        foreach (EntityEntry ee in unchangedEntities)
        {
            ee.State = EntityState.Modified;
        }
        await _appDbContext.SaveChangesAsync();
```

The code above could not track when an element is deleted from the collection and reconcile. While searching for GraphDiff for EFCore, I bumped into Reconcile. 

Reconciler handled my edited and deleted entities but when a new entity is added, then the exception comes knocking.

While debugging my angular request, I noticed my new record needed to have the primary id key of the parent entity. I didn't have to do this prior to using reconciler. Now everything works and I learnt a few lessons too.
